### ISSUE TITLE
url_title doesn't handle diacritics

### ISSUE THREAD

**Issue by SteeveDroz** [codeigniter4/CodeIgniter4#2323]
**Describe the bug**
When passing a string with diacritics to `url_title`, it doesn't handle diacritics in any way.

**CodeIgniter 4 version**
v4.0.0-rc.2.1

**Affected module(s)**
URL helper

**Expected behavior, and steps to reproduce if appropriate**

### Currently

```php
url_title('éléphant de PHP'); // returns "éléphant-de-PHP"
url_title('Éléphant de PHP', '-', true); // returns "Éléphant-de-php"
```

### Expected behaviour

```php
url_title('éléphant de PHP'); // returns "elephant-de-PHP"
url_title('Éléphant de PHP', '-', true); // returns "elephant-de-php"

// or, if retrocompatibility must be kept in some way, add a parameter

url_title('éléphant de PHP', '-', false, true); // returns "elephant-de-PHP"
url_title('Éléphant de PHP', '-', true, true); // returns "elephant-de-php"
```
### Note

There is already a [config file](https://github.com/codeigniter4/CodeIgniter4/blob/develop/system/Config/ForeignCharacters.php) with the list of diacritics.

**Context**
 - OS: Linux Fedora 30
 - Web server Apache 2.4.41
 - PHP version 7.3.10

**Comment by michalsn** [comment]
I think this behavior is ported from CI3 and in such cases we should use `text` helper and it's `convert_accented_characters()` function.

I don't know if there are any contraindications, but I'd love to see this function that converts accented characters also inside the `url_title()` function. I usually create my own "url_title" function just to use those two functions together easier.

**Comment by MGatner** [comment]
I like the idea, but diacritics are also legal URL characters and in some locales may be necessary to distinguish one word from another. I think the original issue is an unexpected return (capital diacritics not returning their equivalent lower case). I hate to add yet another parameter - is it that inconvenient to call both functions when that's the desired return?

**Comment by michalsn** [comment]
After some thoughts - yet another parameter added to this function probably won't change much because it will be added at the end... so I guess calling both functions separately is the right choice.

**Comment by MGatner** [comment]
Fixed by #2325
